src/discof/replay/fd_exec.h

```diff
@@ -0,0 +1,220 @@
+#ifndef HEADER_fd_src_discof_replay_fd_exec_h
+#define HEADER_fd_src_discof_replay_fd_exec_h
+
+#include "../../flamenco/fd_flamenco_base.h"
+#include "../../flamenco/runtime/context/fd_exec_epoch_ctx.h"
+#include "../../flamenco/runtime/context/fd_exec_slot_ctx.h"
+#include "../../flamenco/runtime/fd_runtime_public.h"
+#include "../../flamenco/stakes/fd_stakes.h"
+#include "../../flamenco/runtime/sysvar/fd_sysvar_epoch_schedule.h"
+
+/* Replay tile msg link formatting. The following take a pointer into
+   a dcache region and formats it as a specific message type. */
+
+static inline ulong
+generate_stake_weight_msg( fd_exec_slot_ctx_t * slot_ctx,
+                           fd_spad_t          * runtime_spad,
+                           ulong   epoch,
+                           ulong * stake_weight_msg_out ) {
+  fd_epoch_bank_t * epoch_bank = fd_exec_epoch_ctx_epoch_bank( slot_ctx->epoch_ctx );
+
+  fd_stake_weight_t * stake_weights     = (fd_stake_weight_t *)&stake_weight_msg_out[5];
+  ulong               stake_weight_idx  = fd_stake_weights_by_node( &slot_ctx->slot_bank.epoch_stakes,
+                                                                    stake_weights,
+                                                                    runtime_spad );
+
+  stake_weight_msg_out[0] = epoch;
+  stake_weight_msg_out[1] = stake_weight_idx; /* staked_cnt */
+  stake_weight_msg_out[2] = fd_epoch_slot0( &epoch_bank->epoch_schedule, stake_weight_msg_out[0] ); /* start_slot */
+  stake_weight_msg_out[3] = epoch_bank->epoch_schedule.slots_per_epoch; /* slot_cnt */
+  stake_weight_msg_out[4] = 0UL; /* excluded stake */
+
+  return 5*sizeof(ulong) + (stake_weight_idx * sizeof(fd_stake_weight_t));
+}
```

Comment:
good idea to move this out. can prob also make the original code a lot better, eg. make a `packed` struct for the 5 ulongs that can just be `memcpy` into the struct